### PR TITLE
Update all cloud provider components for Kubernetes 1.31

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR updates all cloud provider components (CCMs and CSIs) to their latest versions and to support Kubernetes 1.31:

- Update vSphere CSI driver to v3.3.1
- Update vSphere CCM to latest patch versions
- Update OpenStack Cinder CSI to latest patch versions
- Update OpenStack CCM to latest patch versions
- Update GCP CSI driver to v1.15.0
- Update Nutanix CSI driver to v2.6.10
- Update Nutanix CCM to v0.4.1
- Update Equinix Metal CCM to v3.8.1
- Update Hetzner CSI to v2.9.0
- Update Hetzner CCM to v1.20.0
- Update DigitalOcean CSI to v4.12.0
- Update DigitalOcean CCM to v0.1.56
- Update AzureFile CSI driver to v1.30.5
- Update AzureDisk CSI driver to v1.30.3
- Update Azure CCM and CNM to latest patch versions
- Update AWS EBS CSI driver to v1.35.0
- Update AWS CCM to v1.31.1

CSI snapshotter has been updated to v8.1.0, machine-controller to v1.60.0, and operating-system-manager to v1.6.0.

**Which issue(s) this PR fixes**:
xref #3347 

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

Testing will be done as part of the testing phase.

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
- Update machine-controller to v1.60.0
- Update operating-system-manager to v1.6.0
- Update CSI snapshotter to v8.1.0
- Update vSphere CSI driver to v3.3.1
- Update vSphere CCM to latest patch versions
- Update OpenStack Cinder CSI to latest patch versions
- Update OpenStack CCM to latest patch versions
- Update GCP CSI driver to v1.15.0
- Update Nutanix CSI driver to v2.6.10
- Update Nutanix CCM to v0.4.1
- Update Equinix Metal CCM to v3.8.1
- Update Hetzner CSI to v2.9.0
- Update Hetzner CCM to v1.20.0
- Update DigitalOcean CSI to v4.12.0
- Update DigitalOcean CCM to v0.1.56
- Update AzureFile CSI driver to v1.30.5
- Update AzureDisk CSI driver to v1.30.3
- Update Azure CCM and CNM to latest patch versions
- Update AWS EBS CSI driver to v1.35.0
- Update AWS CCM to v1.31.1
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

**Review by xmudrii — COMMENTED** [review]
Some comments to help the review process. :)

File: addons/csi-aws-ebs/aws-csi.yaml [hunk 1]
```diff
@@ -679,7 +679,7 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: spec.nodeName
-            - name: EC2_ENDPOINT
+            - name: AWS_EC2_ENDPOINT
```
Comment (xrstf) [inline addons/csi-aws-ebs/aws-csi.yaml:682]: Is this intended? https://github.com/kubermatic/kubeone/tree/main/addons/csi-aws-ebs mentions removing AWS_ env vars.

File: addons/csi-azurefile/csi-azurefile.yaml [hunk 2]
```diff
@@ -615,10 +615,6 @@ spec:
             requests:
               cpu: 10m
               memory: 20Mi
-          securityContext:
-            capabilities:
-              drop:
-              - ALL
```
[the remark this answers is not in the document]
Comment (xrstf) [reply addons/csi-azurefile/csi-azurefile.yaml:621]: They... didn't.

[the remark this answers is not in the document]
Comment (xrstf) [reply addons/csi-azurefile/csi-azurefile.yaml:621]: No they did. God their repo is so confusing :D

File: addons/csi-aws-ebs/aws-csi.yaml (hunk 1 above)
Comment (xmudrii) [reply addons/csi-aws-ebs/aws-csi.yaml:682]: There's zero context on why we changed this in the first place. If tests are passing with `AWS_`, we'll just roll on with that and drop that comment.

File: addons/csi-azurefile/csi-azurefile.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (xrstf) [reply addons/csi-azurefile/csi-azurefile.yaml:621]: However the securityContext is present in the CSI driver's master branch, so soon we won't have to maintain it ourselves.

[the remark this answers is not in the document]
Comment (xmudrii) [reply addons/csi-azurefile/csi-azurefile.yaml:621]: No, it doesn't fall under that clause, it basically covers only that -- CA certs and images. Let's drop it for now, and then add it if they ever add it back to a stable release.

File: pkg/templates/images/images.go [hunk 3]
```diff
@@ -225,53 +225,53 @@ func optionalResources() map[Resource]map[string]string {
...
 
 		// AzureFile CSI driver
 		AzureFileCSI:                   {"*": "mcr.microsoft.com/oss/kubernetes-csi/azurefile-csi:v1.30.5"},
 		AzureFileCSILivenessProbe:      {"*": "mcr.microsoft.com/oss/kubernetes-csi/livenessprobe:v2.13.1"},
 		AzureFileCSINodeDriverRegistar: {"*": "mcr.microsoft.com/oss/kubernetes-csi/csi-node-driver-registrar:v2.12.0"},
 		AzureFileCSIProvisioner:        {"*": "mcr.microsoft.com/oss/kubernetes-csi/csi-provisioner:v5.1.0"},
-		AzureFileCSIResizer:            {"*": "mcr.microsoft.com/oss/kubernetes-csi/csi-resizer:v1.10.1"},
-		AzureFileCSISnapshotter:        {"*": "registry.k8s.io/sig-storage/csi-snapshotter:v8.0.1"}, // use non-MCR image until 8.x is mirrored in MCR
+		AzureFileCSIResizer:            {"*": "mcr.microsoft.com/oss/kubernetes-csi/csi-resizer:v1.11.1"},
+		AzureFileCSISnapshotter:        {"*": "registry.k8s.io/sig-storage/csi-snapshotter:v8.1.0"},
```
Comment (xrstf) [inline pkg/templates/images/images.go:265]: This image _would_ be available in MCR.. for consistency, shouldn't we use it?

Comment (xmudrii) [reply pkg/templates/images/images.go]: It is available, I changed so that we use it for Azure.

**Pull request merged**